pytorch_lightning/trainer/trainer.py

```diff
@@ -447,21 +447,55 @@ def default_attributes(cls):
 
         return args
 
+    @classmethod
+    def get_init_arguments_and_types(cls) -> List[Tuple[str, Set, Any]]:
+        r"""Scans the Trainer signature and returns argument names, types and default values.
+
+        Returns:
+            List with tuples of 3 values:
+            (argument name, set with argument types, argument default value).
+        """
```

Comment:
```suggestion
        
        Examples:
            >>> args = Trainer.get_init_arguments_and_types()
            >>> import pprint
            >>> pprint.pprint(sorted(args))  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
            [('accumulate_grad_batches',
              {<class 'int'>, typing.Dict[int, int], typing.List[list]},
              1),
             ...
             ('callbacks', {<class 'pytorch_lightning.callbacks.base.Callback'>}, []),
             ('check_val_every_n_epoch', {<class 'int'>}, 1),
             ...
             ('max_epochs', {<class 'int'>}, 1000),
             ...
             ('precision', {<class 'int'>}, 32),
             ('print_nan_grads', {<class 'bool'>}, False),
             ('process_position', {<class 'int'>}, 0),
             ('profiler',
              {<class 'pytorch_lightning.profiler.profiler.BaseProfiler'>,
               <class 'NoneType'>},
              None),
            ...
        """
```
this works fine

Reply:
but it revies that it also gives deprecated arguments, then we shall add then to some kind of ignore list